src/pac.c

```diff
@@ -124,13 +144,47 @@ pac_alloc_real(tsdn_t *tsdn, pac_t *pac, ehooks_t *ehooks, size_t size,
 		edata = ecache_alloc(tsdn, pac, ehooks, &pac->ecache_muzzy,
 		    NULL, size, alignment, zero, guarded);
 	}
+#ifdef LIMIT_USIZE_GAP
+	/*
+	 * Split is not supported when (!map_coalesce && !opt_retain).
+	 * Do not try to bump the size for this case.
+	 */
+	if (sz_limit_usize_gap_enabled() && edata == NULL &&
+	    (maps_coalesce || opt_retain)) {
+		size_t caching_size = pac_batched_alloc_retained_size(
```

Comment:
caching_size is a bit confusing. Maybe `expanded_size` or `batched_size`?

Reply:
Changed to `batched_size`, I think this name explains itself best.